src/transformers/models/musicgen/modeling_musicgen.py

```diff
@@ -773,7 +773,7 @@ def forward(
         past_key_values_length = past_key_values[0][0].shape[2] if past_key_values is not None else 0
 
         if inputs_embeds is None:
-            inputs_embeds = torch.zeros((bsz, seq_len, self.d_model), device=input_ids.device)
+            inputs_embeds = torch.zeros((bsz, seq_len, self.d_model), dtype=self.dtype, device=input_ids.device)
```

Comment:
I am not sure about this, but @sgugger knows everything.

Reply:
This will take the dtype of a random weight (not really random but the first one). Might be better to look for the dtype of the embeddings instead, no?

Reply:
Refactored to avoid having to specify any dtype / device arguments: https://github.com/huggingface/transformers/pull/25169/commits/6118fbf561fe483ed1c3b6442c8fe8a2ec0d9411